### PR TITLE
[Transform] Add ToLevi transform

### LINKED ISSUE #4613: Add toLevi transform
## 🔨Work Item

Provide a toLevi transform to convert a graph to a corresponding Levi graph (convert original edges into nodes in the new graph).

Such operation is widely used in graph2seq applications.

**IMPORTANT:**
* This template is only for dev team to track project progress. For feature request or bug report, please use the corresponding issue templates.
* DO NOT create a new work item if the purpose is to fix an existing issue or feature request. We will directly use the issue in the project tracker.

Project tracker: https://github.com/orgs/dmlc/projects/2

### LINKED ISSUE #4883: [Bug] extract_edge_subframes has a typo with parameter edges_or_device
## 🐛 Bug

The parameters passed into `extract_edge_subframes` are `graph`, `edges_or_device` and `store_ids`, but it uses `nf.to(device)` which should be `nf.to(edges_or_device)`, and it will raise an exception with `"device" is not defined` when passing a device object by `edges_or_device`.

## Expected behavior

Line 958 in python/dgl/utils/internal.py should be `edge_frames = [nf.to(edges_or_device) for nf in graph._edge_frames]`

### PR DESCRIPTION
**Pull request by ZHITENGLI**
## Description

close #4613 
fix #4883 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

## Changes

Provide a toLevi transform to convert a graph to a corresponding Levi graph (convert original edges into nodes in the new graph).
Fix parameter typo in `extract_edge_subframes`.

### CODE REVIEW

**Comment by rudongyu** [comment]
@dgl-bot

**Comment by rudongyu** [comment]
@dgl-bot

File: python/dgl/transforms/module.py [hunk 1]
```diff
@@ -1718,3 +1720,67 @@ def ppr(self, g):
...
+        Parameters
+        ----------
+        g : DGLGraph
+            The input graph, should be a homogeneous directed graph.
+
+        Returns
+        -------
+        DGLGraph
+            The Levi graph of input, will be a heterogeneous graph, where ntypes 'node' and
+            'edge' have the corresponding ids to nodes and edges in the original graph.
```
Comment (rudongyu) [inline python/dgl/transforms/module.py:1769]:
```suggestion
            The Levi graph of input, will be a heterogeneous graph, where nodes of ntypes ``'node'`` and
            ``'edge'`` have corresponding IDs of nodes and edges in the original graph.
```

File: python/dgl/transforms/module.py [hunk 2]
```diff
@@ -1718,3 +1720,67 @@ def ppr(self, g):
...
+        n2e = edge_list[0], F.arange(0, g.num_edges(), idtype, device)
+        e2n = F.arange(0, g.num_edges(), idtype, device), edge_list[1]
+        graph_data = {('node', 'n2e', 'edge'): n2e,
+                      ('edge', 'e2n', 'node'): e2n}
+        levi_g = convert.heterograph(graph_data, idtype=idtype, device=device)
+
+        # copy ndata and edata
+        node_frames = utils.extract_node_subframes(g, nodes_or_device=device)
+        edge_frames = utils.extract_edge_subframes(g, edges_or_device=device)
+        utils.set_new_frames(levi_g, node_frames=edge_frames+node_frames)
```
Comment (rudongyu) [inline python/dgl/transforms/module.py:1784]: It uses the internal property of node types that they are ordered in alphabetic order. Maybe add a comment above to make it clear.

File: python/dgl/transforms/module.py [hunk 3]
```diff
@@ -1718,3 +1720,67 @@ def ppr(self, g):
...
+        idtype = g.idtype
+
+        edge_list = g.edges()
+        n2e = edge_list[0], F.arange(0, g.num_edges(), idtype, device)
+        e2n = F.arange(0, g.num_edges(), idtype, device), edge_list[1]
+        graph_data = {('node', 'n2e', 'edge'): n2e,
+                      ('edge', 'e2n', 'node'): e2n}
+        levi_g = convert.heterograph(graph_data, idtype=idtype, device=device)
+
+        # copy ndata and edata
```
Comment (rudongyu) [inline python/dgl/transforms/module.py:1781]: I recommend to add a functional API and add an optional argument for copy_data. What do you think?

**Comment by rudongyu** [comment]
@dgl-bot

File: python/dgl/transforms/module.py (hunk 3 above)
Comment (ZHITENGLI) [reply python/dgl/transforms/module.py:1781]: If we don't copy features, we cannot use message passing paradigm in Levi graph. That doesn't seem to make sense.

**Comment by rudongyu** [comment]
@dgl-bot

**Review by rudongyu — APPROVED** [review]
LGTM

**Pull request merged**